src/context/AuthContext.tsx

```diff
@@ -0,0 +1,71 @@
+import { createContext, useContext, useState, useEffect } from 'react';
+import type { ReactNode } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+interface User {
+  name: string;
+  email: string;
+}
+
+interface AuthContextType {
+  isLoggedIn: boolean;
+  user: User | null;
+  login: (email: string) => void;
+  logout: () => void;
+}
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
```

Comment:
isLoggedIn 상태가 꼭 필요할까요??

user 상태를 기준으로 계산해도 문제가 없어보여서요!
```suggestion
  const isLoggedIn = !!user;
```

Reply:
아직 !! 연산자가 조금 어색해서 로그인 상태를 별도로 관리했었는데,
말씀해주신 것처럼 !!user를 활용하면 상태도 줄고 코드도 훨씬 간결해지는 것 같습니다!
더 익숙해지도록 연습해보겠습니다. 감사합니다